apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java

```diff
@@ -651,6 +657,88 @@ public void setParent(@UriVariable final String space, @UriVariable final String
 
     }
 
+    @ApiOperation(value = "Retrieves all comments for a given node")
+    @Uri(value = "/nodes/{space}/{store}/{guid}/comments", method = HttpMethod.GET)
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 403, message = "Not Authorized"),
+            @ApiResponse(code = 404, message = "Not Found")
+    })
+    public void getComments(@UriVariable String space, @UriVariable String store, @UriVariable String guid,
+            @RequestParam int skipcount, @RequestParam int pagesize, WebScriptResponse response) throws IOException {
+        final NodeRef target = new NodeRef(space, store, guid);
+        if (nodeService.exists(target)) {
+            if(permissionService.hasPermission(target, PermissionService.READ)) {
+                Conversation comments = commentService.getComments(target, skipcount, pagesize);
+                boolean canCreate = permissionService.hasPermission(target, PermissionService.CREATE_CHILDREN);
+                comments.setCreatable(canCreate);
+                writeJsonResponse(response, comments);
+            } else {
+                throw new AccessDeniedException("User does not have permission to read parent node");
+            }
+        } else {
+            writeNotFoundResponse(response, target);
+        }
+    }
+
+    @ApiOperation(value = "Appends a new comment to the given node.")
+    @Uri(value = "/nodes/{space}/{store}/{guid}/comments", method = HttpMethod.POST)
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 403, message = "Not Authorized"),
+            @ApiResponse(code = 404, message = "Not Found")
+    })
+    public void postComment(@UriVariable String space, @UriVariable String store, @UriVariable String guid,
```

Comment:
(non-blocking) If the other method is called `updateComment`, this one should be called `createComment` or `addComment` IMO.